src/dc_mimeparser.rs

```diff
@@ -127,8 +127,12 @@ impl<'a> MimeParser<'a> {
             if let Some(field) = self.lookup_field("Subject") {
                 if (*field).fld_type == MAILIMF_FIELD_SUBJECT as libc::c_int {
                     let subj = (*(*field).fld_data.fld_subject).sbj_value;
-
-                    self.subject = as_opt_str(subj).map(dc_decode_header_words);
+                    let subj = to_opt_string_lossy(subj);
+                    self.subject = if subj.is_some() {
+                        Some(dc_decode_header_words(&subj.unwrap()))
+                    } else {
+                        None
```

Comment:
recomend to leave the map -- it means apply dc_decode_header_words(x) if were a Some(x), or assume None. 
In other words map() is only applied to Some contents.

Reply:
it is a bit more tricky as dc_decode_header_words() takes a `&str` where subj is now a `String`. with some rust-magic, map() will probably still work, though.

Reply:
you can still use a closure `.map(|x| dc_decode_header_words(&x))`